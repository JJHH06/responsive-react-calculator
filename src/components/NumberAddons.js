import React, {useEffect, useState} from 'react'

const NumberAddons = ({operacion,addonOperator}) =>{
    const clickOperatorListener = e =>{
        addonOperator(operacion);
    }
    return (
        <div className = "col calculator-numbers" onClick = {clickOperatorListener}>
            <p className = "calculator-appenders">{operacion}</p>
        </div>
    )
}

export default NumberAddons