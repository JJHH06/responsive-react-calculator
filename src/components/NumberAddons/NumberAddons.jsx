/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
// se desactivaron las reglas que no dejan que funcionen los botones
import React from 'react';
import PropTypes from 'prop-types';

const NumberAddons = ({ operacion, addonOperator }) => {
  const clickOperatorListener = () => {
    addonOperator(operacion);
  };
  return (
    <div className="col calculator-numbers" onClick={clickOperatorListener}>
      <p className="calculator-appenders">{operacion}</p>
    </div>
  );
};

NumberAddons.propTypes = {
  operacion: PropTypes.string.isRequired,
  addonOperator: PropTypes.func.isRequired,
};

export default NumberAddons;
