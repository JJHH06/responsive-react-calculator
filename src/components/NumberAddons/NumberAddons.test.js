/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import NumberAddons from './NumberAddons';

afterEach(cleanup);
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<NumberAddons />, div);
});

it('renders NumberAddOn correctly', () => {
  const { getByTestId } = render(<NumberAddons operacion="." testid="operator." />);
  expect(getByTestId('operator.')).toHaveTextContent('.');
});

it('renders second NumberAddOn correctly', () => {
  const { getByTestId } = render(<NumberAddons operacion="=" testid="operator=" />);
  expect(getByTestId('operator=')).toHaveTextContent('=');
});

it('matches snapshot', () => {
  const tree = renderer.create(<NumberAddons operacion="=" testid="operator=" />).toJSON();
  expect(tree).toMatchSnapshot();
});
