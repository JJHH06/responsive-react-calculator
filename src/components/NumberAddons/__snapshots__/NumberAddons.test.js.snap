// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="col calculator-numbers"
  data-testid="operator="
  onClick={[Function]}
>
  <p
    className="calculator-appenders"
  >
    =
  </p>
</div>
`;
